crates/oxc_linter/src/rules/jest/no_deprecated_functions.rs

```diff
@@ -0,0 +1,173 @@
+    /// These functions were later moved onto the `jest` object in order to be easier
+    /// for type checkers to handle, and their use via `require` deprecated. Finally,
+    /// the release of Jest 26 saw them removed from the `require` function altogether.
+    ///
+    /// #### `jest.runTimersToTime`
+    /// This function was renamed to `advanceTimersByTime` in Jest 22 and removed in Jest 27.
+    ///
+    /// #### `jest.genMockFromModule`
+    /// This function was renamed to `createMockFromModule` in Jest 26, and is scheduled for removal in Jest 30.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// While typically these deprecated functions are kept in the codebase for a number
+    /// of majors, eventually they are removed completely.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// jest.resetModuleRegistry // since Jest 15
+    /// jest.addMatchers // since Jest 17
+    /// ```
+    NoDeprecatedFunctions,
+    style,
+);
+
+const DEPRECATED_FUNCTIONS_MAP: Map<&'static str, (usize, &'static str)> = phf_map! {
+    "jest.resetModuleRegistry" => (15, "jest.resetModules"),
+    "jest.addMatchers" => (17, "expect.extend"),
+    "require.requireMock" => (21, "jest.requireMock"),
+    "require.requireActual" => (21, "jest.requireMock"),
+    "jest.runTimersToTime" => (22, "jest.advanceTimersByTime"),
+    "jest.genMockFromModule" => (26, "jest.createMockFromModule"),
+};
+
+impl Rule for NoDeprecatedFunctions {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let version = value
+            .get(0)
+            .and_then(|v| v.get("jest"))
+            .and_then(|v| v.get("version"))
+            .and_then(|v| serde_json::Value::as_str(v))
+            .unwrap();
+
+        Self { jest: JestConfig { version: version.to_string() } }
+    }
+
+    fn run<'a>(&self, node: &oxc_semantic::AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::MemberExpression(mem_expr) = node.kind() else {
+            return;
+        };
+        let mut chain: Vec<Cow<'a, str>> = Vec::new();
+        chain.extend(get_node_name_vec(mem_expr.object()));
+
+        if let Some(name) = mem_expr.static_property_name() {
+            chain.push(Cow::Borrowed(name));
+        }
+
+        let node_name = chain.join(".");
+        let major: Vec<&str> = self.jest.version.split('.').collect();
+        let jest_version_num: usize = major[0].parse().unwrap();
+
+        if let Some((base_version, replacement)) = DEPRECATED_FUNCTIONS_MAP.get(&node_name) {
+            if jest_version_num >= *base_version {
+                ctx.diagnostic_with_fix(
+                    DeprecatedFunction(node_name, (*replacement).to_string(), mem_expr.span()),
+                    || Fix::new(*replacement, mem_expr.span()),
+                );
+            }
+        }
+    }
+}
+
+#[test]
+fn tests() {
+    use crate::tester::Tester;
+
+    let pass = vec![
+        ("jest", Some(serde_json::json!([{ "jest": { "version": "14" } }]))),
```

Comment:
The original implementation read Jest version from [`context.setting`](https://github.com/jest-community/eslint-plugin-jest/blob/a058f22f94774eeea7980ea2d1f24c6808bf3e2c/src/rules/no-deprecated-functions.ts#L45-L48) first, I have no idea how to port this 🙃

Reply:
I think currently we can target a specify version first, such as `14` or `29`, and latter fix it through `cli` option or related config file.